fireshape/innerproduct.py

```diff
@@ -183,10 +198,9 @@ class InterpolatedInnerProductImpl(InnerProductImpl):
     zero-rows with rows that have 1 on the diagonal entry. Finally, create a
     ksp.solver.
     """
-
-    def __init__(self, inner_product, V_r, I):
-        A = inner_product.get_impl(V_r).A
-        ITAI = A.PtAP(I)
+    def get_impl(self, V, I):
```

Comment:
Why do we have the multiplication with the interpolation matrix at two locations now?

Reply:
InterpolatedInnerProduct is not necessary anymore and will disappear in the next commit, give me 2mins

Reply:
the advantages of removing InterpolatedInnerProduct and merging it it Innerproducts are that:
1- the type of ControlSpace.inner_product is indeed inner_product
2- the code to create BsplineControl.inner_product is essentially the same as FEControlSpace